engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -71,6 +71,14 @@ public ComponentSystemManager(Context context) {
         this.context = context;
     }
 
+    private boolean ifRegister(RegisterSystem registerSystem, ModuleEnvironment environment) {
+        if (registerSystem.requiresOptional().length == 0) return true;
+        for (String moduleName :registerSystem.requiresOptional()) {
+            if(environment.get(new Name(moduleName)) == null) return false;
+        }
+        return true;
+    }
+
```

Comment:
Could need a more expressive name and some formatting updates :)

```
    private boolean areOptionalRequirementsContained(RegisterSystem registerSystem, ModuleEnvironment environment) {
        if (registerSystem.requiresOptional().length == 0) {
            return true;
        }
        for (String moduleName : registerSystem.requiresOptional()) {
            if (environment.get(new Name(moduleName)) == null) {
                return false;
            }
        }
        return true;
    }
```

It is also a common pratice to define helper methods below their call so that a reader of the class can scroll downwards instead of jumping back and forth.
Therefore this should be below `loadSystems`